tests/epyccel/test_generic_functions.py

```diff
@@ -95,13 +94,341 @@ def test_tmplt_tmplt_1(language):
+        )
+    )
+)
+def test_float_types(language):
+    f1 = epyccel(mod2.float_types , language = language, verbose=True)
+    f2 = mod2.float_types
+
+    assert f1(10.5, 5.5) == f2(10.5, 5.5)
+    assert f1(np.float(15.5) , np.float(10.5)) == f2(np.float(15.5) , np.float(10.5))
+    assert f1(np.float32(155.2), np.float32(177.1)) == f2(np.float32(155.2), np.float32(177.1))
+    assert f1(np.float64(166.6), np.float64(255.6)) == f2(np.float64(166.6), np.float64(255.6))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(reason='issue #567 and 568: Duplicated types in a template and header'),
+            pytest.mark.fortran]
+        )
+    )
+)
+def test_complex_types(language):
+    f1 = epyccel(mod2.complex_types , language = language, verbose=True)
+    f2 = mod2.complex_types
+
+    assert f1(complex(1, 2.2), complex(1, 2.2)) == f2(complex(1, 2.2), complex(1, 2.2))
+    assert f1(np.complex(15.5, 2.0) , np.complex(10.5, 3.4)) == f2(np.complex(15.5, 2.0) , np.complex(10.5, 3.4))
+    assert f1(np.complex64(15.5 + 2.0j) , np.complex64(10.5 + 3.4j)) == f2(np.complex64(15.5 + 2.0j) , np.complex64(10.5 + 3.4j))
+    assert f1(np.complex128(15.5+ 2.0j) , np.complex(10.5+ 3.4j)) == f2(np.complex128(15.5+ 2.0j) , np.complex(10.5+ 3.4j))
+
+def test_mix_types_1(language):
+    f1 = epyccel(mod2.mix_types_1 , language = language, verbose=True)
+    f2 = mod2.mix_types_1
+
+    assert f1(complex(1, 2), 15, np.int16(5)) == f2(complex(1, 2), 15, np.int16(5))
+    assert f1(complex(1, 2), 15, True) == f2(complex(1, 2), 15, True)
+    assert f1(complex(1, 2), 7.0, np.int16(5)) == f2(complex(1, 2), 7.0, np.int16(5))
+    assert f1(complex(1, 2), 7.0, False) == f2(complex(1, 2), 7.0, False)
+    assert f1(15, 14, np.int16(2012)) == f2(15, 14, np.int16(2012))
+    assert f1(15, 14, True) == f2(15, 14, True)
+    assert f1(15, 7.0, np.int16(2012)) == f2(15, 7.0, np.int16(2012))
+    assert f1(15, 14, False) == f2(15, 14, False)
+
+
+def test_mix_types_2(language):
+    f1 = epyccel(mod2.mix_types_2 , language = language, verbose=True)
+    f2 = mod2.mix_types_2
+
+    assert f1(-1, -1) == f2(-1, -1)
+    assert f1(np.int64(4), np.int64(16)) == f2(np.int64(4), np.int64(16))
+    assert f1(np.int16(4), np.int16(4)) == f2(np.int16(4), np.int16(4))
+    assert f1(5.7, -1.2) == f2(5.7, -1.2)
+    assert f1(complex(7.2, 3.12), complex(7.2, 3.12)) == f2(complex(7.2, 3.12), complex(7.2, 3.12))
+    assert f1(np.float32(16), np.float32(16)) == f2(np.float32(16), np.float32(16))
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("fortran", marks = [
+            pytest.mark.xfail(sys.platform == 'win32', reason='issue #567 and #568: Duplicated types in a template and header'),
+            pytest.mark.fortran]
+        )
+    )
+)
+def test_mix_types_3(language):
```

Comment:
After your last modification this will also fail on all platforms

Reply:
this one should only fail in fortran true

Reply:
c win32